pxf/pxf-json/src/main/java/org/apache/hawq/pxf/plugins/json/JsonResolver.java

```diff
@@ -0,0 +1,256 @@
+					addFieldFromJsonArray(columnMetadata.getColumnType(), node, columnMetadata.getArrayNodeIndex());
+				} else {
+					throw new IllegalStateException(columnMetadata.getColumnName() + " is not an array node");
+				}
+			} else {
+				// This column is not an array type
+				// Add the value to the record
+				addFieldFromJsonNode(columnMetadata.getColumnType(), node);
+			}
+		}
+
+		return oneFieldList;
+	}
+
+	/**
+	 * @return Returns a row comprised of typed, empty fields. Used as a result of broken/malformed json records.
+	 */
+	private List<OneField> createEmptyRow() {
+		ArrayList<OneField> emptyFieldList = new ArrayList<OneField>();
+		for (ColumnDescriptorCache column : columnDescriptorCache) {
+			emptyFieldList.add(new OneField(column.getColumnType().getOID(), null));
+		}
+		return emptyFieldList;
+	}
+
+	/**
+	 * Iterates down the root node to the child JSON node defined by the projs path.
+	 * 
+	 * @param root
+	 *            node to to start the traversal from.
+	 * @param projs
+	 *            defines the path from the root to the desired child node.
+	 * @return Returns the child node defined by the root and projs path.
+	 */
+	private JsonNode getChildJsonNode(JsonNode root, String[] projs) {
+
+		// Iterate through all the tokens to the desired JSON node
+		JsonNode node = root;
+		for (int j = 0; j < projs.length; ++j) {
+			node = node.path(projs[j]);
+		}
+
+		return node;
+	}
+
+	/**
+	 * Iterates through the given JSON node to the proper index and adds the field of corresponding type
+	 * 
+	 * @param type
+	 *            The {@link DataType} type
+	 * @param node
+	 *            The JSON array node
+	 * @param index
+	 *            The array index to iterate to
+	 * @throws IOException
+	 */
+	private void addFieldFromJsonArray(DataType type, JsonNode node, int index) throws IOException {
+
+		int count = 0;
+		boolean added = false;
+		for (Iterator<JsonNode> arrayNodes = node.getElements(); arrayNodes.hasNext();) {
+			JsonNode arrayNode = arrayNodes.next();
+
+			if (count == index) {
+				added = true;
+				addFieldFromJsonNode(type, arrayNode);
+				break;
+			}
+
+			++count;
+		}
+
+		// if we reached the end of the array without adding a field, add null
+		if (!added) {
+			addNullField(type);
+		}
+	}
+
+	/**
+	 * Adds a field from a given JSON node value based on the {@link DataType} type.
+	 * 
+	 * @param type
+	 *            The DataType type
+	 * @param val
+	 *            The JSON node to extract the value.
+	 * @throws IOException
+	 */
+	private void addFieldFromJsonNode(DataType type, JsonNode val) throws IOException {
+		OneField oneField = new OneField();
+		oneField.type = type.getOID();
+
+		if (val.isNull()) {
+			oneField.val = null;
+		} else {
+			switch (type) {
+			case BIGINT:
+				oneField.val = val.asLong();
+				break;
+			case BOOLEAN:
+				oneField.val = val.asBoolean();
+				break;
+			case CHAR:
+				oneField.val = val.asText().charAt(0);
+				break;
+			case BYTEA:
```

Comment:
As far as bytea is not tested and isn't supported, should we remove this section?